packages/qwik/src/core/use/use-context.ts

```diff
@@ -263,32 +267,39 @@ export interface UseContext {
  * ```
  *
  * @param context - The context to retrieve a value from.
+ * @param defaultValue - The default value to be used if value is missing.
  * @public
  */
 // </docs>
-export const useContext: UseContext = <STATE extends object>(
+export const useContext: UseContext = <STATE>(
   context: ContextId<STATE>,
-  defaultValue?: any
+  defaultValue?: STATE | DefaultStateFn<STATE> // type of default value should be the same as the context
 ) => {
   const { val, set, iCtx, elCtx } = useSequentialScope<STATE>();
+
   if (val !== undefined) {
     return val;
   }
+
   if (qDev) {
     validateContext(context);
   }
 
   const value = resolveContext(context, elCtx, iCtx.$renderCtx$.$static$.$containerState$);
+
   if (typeof defaultValue === 'function') {
-    return set(invoke(undefined, defaultValue, value));
+    return set(invoke(undefined, defaultValue as DefaultStateFn<STATE>, value));
   }
+
   if (value !== undefined) {
     return set(value);
   }
+
   if (defaultValue !== undefined) {
     return set(defaultValue);
   }
-  throw qError(QError_notFoundContext, context.id);
+
+  throw missingContextValueError(context.id);
```

Comment:
Please inline the function here, no need to make a short function that is used only from one place.